### ISSUE TITLE
@properties.defaults wrapper needs an overhaul

### ISSUE THREAD

**Issue by fwkoch** [3ptscience/properties#65]
Right now, this wrapper is used to `add_defaults` to superclass `_defaults` - that should probably be more explicit. - These methods should happen on startup and not use `self`.

There should also be functions to dynamically calculate defaults of properties (something like `@properties.default('prop_name')`.

**Comment by fwkoch** [comment]
Inheriting multiple `defaults`-wrapped `_defaults` functions doesn't actually work.
```
import properties


class A(properties.HasProperties):
    int_0 = properties.Integer('an integer', default=0)


class B(A):
    @properties.defaults
    def _defaults(self):
        return dict(int_0=1)


class C(B):
    @properties.defaults
    def _defaults(self):
        return dict(int_0=2)


a = A()
assert a.int_0 == 0  # True
b = B()
assert b.int_0 == 1  # True
c = C()  # Infinite loop
```
This is because: `super(self.__class__, self)._defaults` is called in `C` then in `B`, but `self` is still `c` so super is `B` again, etc.

Anyway, needs an overhaul.

**Comment by fwkoch** [comment]
Current solution: `@properties.defaults` wrapper removed. To replace this behaviour, `_defaults` is still a dictionary of default values, but it follows class inheritance. One default can be overridden on a class using `_defaults` while others remain unchanged. To allow for dynamic defaults, the values in this dictionary can be callable.

Quick example:
```python
import properties
import random

class A(properties.HasProperties):
    """All defaults 0"""
    a = properties.Integer('int a', default=0)
    b = properties.Integer('int b', default=0)
    c = properties.Integer('int c', default=0)

class B(A):
    """defaults are explicitly overridden"""
    _defaults = {'a': 10, 'b': 20, 'c': 30}

class C(B):
    """default for a is changed to a randomly generated int.
    defaults for b and c are still 20 and 30, respectively.
    """
    _defaults = {'a': lambda: random.randint(0, 10)}
```
This is implemented in #76
